repose-aggregator/src/docs/asciidoc/services/http-connection-pool.adoc

```diff
@@ -1,6 +1,183 @@
+          chunked-encoding="true" <!--3-->
+          http.conn-manager.max-total="400" <!--4-->
+          http.conn-manager.max-per-route="200" <!--5-->
+          http.socket.timeout="30000" <!--6-->
+          http.socket.buffer-size="8192" <!--7-->
+          http.connection.timeout="30000" <!--8-->
+          http.connection.max-line-length="8192" <!--9-->
+          http.connection.max-header-count="100" <!--10-->
+          http.connection.max-status-line-garbage="100" <!--11-->
+          http.tcp.nodelay="true" <!--12-->
+          keepalive.timeout="0" <!--13-->
+          keystore-filename="client.jks" <!--14-->
+          keystore-password="password" <!--15-->
+          key-password="password" <!--16-->
+          truststore-filename="servers.jks" <!--17-->
+          truststore-password="password"> <!--18-->
+        <headers> <!--19-->
+            <header name="foo" value="bar"/> <!--20-->
+        </headers>
+    </pool>
+    <pool id="banana"/> <!--21-->
+</http-connection-pools>
+----
+<1> The id/name of this pool, it must be unique.
+    Used to identify this pool for usage by other services/filters.
+    The only required attribute for a pool.
+<2> Whether or not this pool is the default pool for the connection service.
+    You must have exactly one pool marked as the default.
+    This pool is the one that will be used to make connections to the origin service.
+<3> Determines whether or not this client should send chunked data when forwarding requests to the origin service.
+    If set to true, chunked data will always be sent.
+    If set to false, chunked data will never be sent.
+    If set to auto, chunked data will only be sent if the original request to Repose was chunked.
+    Note that the auto setting only has meaning for the default connection pool, as that is the only pool that will be used to make requests to the origin service. +
+    *WARNING*: Setting this attribute to false may cause *Repose* to attempt to retrieve the actual content length through reading the ServletInputStream.
+    This will cause some performance degradation as request body is no longer always streamed through. +
+    It is recommended that users leave this value as true unless their service does not support chunked encoding. +
+    Default is true.
+<4> Maximum number of connections opened across all endpoints in this connection pool.
+    If this is set too high, you may run out of memory. +
+    Default is 400.
+<5> Maximum number of connections opened per endpoint in this connection pool. +
+    Default is 200.
+<6> Defines the socket timeout in milliseconds, which is the timeout for waiting for data (or, put differently, a maximum period of inactivity between two consecutive data packets).
+    A timeout value of zero is interpreted as an infinite timeout. +
+    Default is 30000.
+<7> Determines the size of the internal socket buffer used to buffer data while receiving / transmitting HTTP messages. +
```

Comment:
Oh man. That trailing `+` caught me off guard since GitHub rendered it on a separate line. I had a "wat" moment. All good now.